roboteam_ai/src/utilities/Field.cpp

```diff
@@ -78,6 +81,153 @@ bool Field::pointIsInField(Vector2 point, float margin) {
 
 }
 
+double Field::getPercentageOfGoalVisibleFromPoint(bool ourGoal, Vector2 point){
+    auto field = Field::get_field();
+    double goalWidth = field.goal_width;
+    double blockadeLength = 0;
+    for (auto const &blockade : getBlockadesMappedToGoal(ourGoal, point)) {
+        blockadeLength += blockade.first.dist(blockade.second);
+    }
+    return std::max(100 - round(blockadeLength/goalWidth * 100), 0.0);
```

Comment:
I like fmax a bit better since you're also using round() from the math.h header for consistency